### ISSUE TITLE
 Unknown option {cat_features}

### ISSUE THREAD

**Issue by Steviey** [catboost/catboost#1445]
Problem:  **Unknown option {cat_features}**
catboost version: 1.24.1
Operating System: Ubuntu 1.16.x LTS
R 4.x
CPU: Yes
**Code:**

```
[...]
myGrid <- expand.grid(
	iterations=c(300)
	,learning_rate = c(0.1)
	,bootstrap_type=c('Bayesian','Bernoulli','MVS','No')
	,cat_features=c(0,1,2,3,4,5)
[...]	

```
```
[...]
if(tMode=='testing'){
	resObj$cvDf <-catboost.cv(allData_pool, params, fold_count = 5,
	type = "Classical", partition_random_seed = 321123, shuffle = TRUE,
	stratified = FALSE, early_stopping_rounds = NULL)
}else{
	resObj$model <- catboost.train(train_pool,test_pool,params)
}
[...]
```
**Errors:**
```
**Case catboost.cv():**
Error in catboost.cv(allData_pool, params, fold_count = 5, type = "Classical",  : 
  catboost/private/libs/options/plain_options_helper.cpp:482: Unknown option {cat_features} with value "5"
**Case catboost.train():**
Error in catboost.train(train_pool, test_pool, params) : 
  catboost/private/libs/options/plain_options_helper.cpp:482: Unknown option {cat_features} with value "5"

```

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Types of features are part of the dataset description, and are not boosting options.
Hence the exception.

**Comment by Steviey** [comment]
@Evgueni-Petrov-aka-espetrov Evgueni

From: https://catboost.ai/docs/concepts/r-training-parameters.html?lang=en

```
cat_features:
A vector of categorical features indices.
The indices are zero-based and can differ from the ones given in the columns description file.
```
There seems to be no example, how to use such a description file in a R project, only a general overview... https://catboost.ai/docs/concepts/input-data_column-descfile.html#input-data_column-descfile
 
Can you provide some example lines of code?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Here is an example of how to use column description files.
Such files specify layouts of datasets which are not R data frames, e.g. files.
Not sure how is.factor() can be of use here.

https://github.com/catboost/catboost/blob/8305d3c0a383c981e6cf08fc32ab9c52acb7fe11/catboost/R-package/tests/testthat/test_on_trimmed_adult_dataset.R#L6

**Comment by Steviey** [comment]
OK, this seems to be the right answer (without external file, including factors)....

https://github.com/catboost/catboost/issues/1246

Have to test it out.

**Comment by Steviey** [comment]
Turns out there is no external file involved at all... I did it before but forgot it.

`train_pool <- catboost.load_pool(data=trainX,label=trainY,cat_features=c(3,4),feature_names = as.list(colnames(trainX)))`
